public/js/controllers/pagenavigation_controller.js

```diff
@@ -21,6 +19,18 @@ export default class extends Controller {
     Turbolinks.visit(url.toString())
   }
 
+  setPageAge () {
+    var url = Url(window.location.href)
+    var q = Url.qs.parse(url.query)
+    q[this.pagesizeTarget.dataset.offsetkey] = this.pageageTarget.selectedOptions[0].value
```

Comment:
The difference between `setPageAge` and  `setPageSize` is this line.

As it is now, this line is overwriting the offset value set in `setPageSize`, if you come up with a new dataset key and value instead of offset you can easily add this line to `setPageSize` and call it instead.